src/silx/io/utils.py

```diff
@@ -854,7 +854,7 @@ def match(group, path_pattern: str) -> Generator[str, None, None]:
                 yield f"{matching_path}/{matching_subpath}"
 
 
-def get_data(url):
+def get_data(url: Union[str,silx.io.url.DataUrl], check_schemas: bool=False):
```

Comment:
Why adding a `check_schemas=False` argument rather than always trying with `silx` and `fabio` schemas?
To me the improvement you did makes sense all the time.

+ `silx.io.url.DataUrl` typing looks to cause trouble to `pytest` loading mechanism...